### ISSUE TITLE
Add ability to send custom symbology for feature server

### ISSUE THREAD

**Issue by jptheripper** [koopjs/FeatureServer#80]
You currently have to over the point.json (or line, fill-symbol, polygon) files in \node_modules\featureserver\templates\renderers\symbology to override FeatureServer symbology.  Please add the ability to pass a custom json file at the provider level.

**Comment by SergeyBarskiy** [comment]
Overwriting templates does not work well if I want to have custom symbol per layer.  Ideally if I set metadata.drawingInfo.renderer the templates.js would leave it along and only set the rendered if it is undefined or null.

**Comment by dmfenton** [comment]
yeah this needs to be set in the geojson.metadata response

**Comment by SergeyBarskiy** [comment]
@dmfenton 
I think this line if (json.drawingInfo) json.drawingInfo.renderer = renderers[json.geometryType]
should be if (json.drawingInfo && ! json.drawingInfo.renderer) json.drawingInfo.renderer = renderers[json.geometryType]

**Comment by hhkaos** [comment]
I'm also struggling with this, so I have changed [templates.js](https://github.com/koopjs/FeatureServer/blob/master/src/templates.js#L53):
```js
if (json.drawingInfo) json.drawingInfo.renderer = renderers[json.geometryType]
```
For:
```js
if (json.drawingInfo && metadata.drawingInfo) json.drawingInfo = metadata.drawingInfo;
```

I copied the drawingInfo field from the webmap spec and it worked perfectly.

**Comment by ButchGis** [comment]
In the provider, I am setting the layer metadata with an additional property: KOOP_SYMBOL.
like so:
metadata.KOOP_SYMBOL = the json for a renderer

Currently, I have a koop provider cycling through a configurable set of point renderers.
Mostly just changing color.  

I had to modify 1 line in FeatureServer.

 At line 42 in template.js I changed, "json.drawingInfo.renderer = renderers[json.geometryType]", to "json.drawingInfo.renderer = metadata.KOOP_SYMBOL || renderers[json.geometryType]"

Maybe a better name of the metadata attribute could be chosen. But this seems to allow the desired functionality of setting symbol per layer at provider.

**Comment by rgwozdz** [comment]
@ButchGis - nice work.  Do you want to PR this?  My only suggestion is `metadata.renderer` instead of `metadata.KOOP_SYMBOL`.

**Comment by ButchGis** [comment]
Sure. After holiday weekend. Thanks.

Get Outlook for Android<https://urldefense.proofpoint.com/v2/url?u=https-3A__aka.ms_ghei36&d=DwIF-g&c=n6-cguzQvX_tUIrZOS_4Og&r=gaXh_UY66bShHyUniiKvr2y8f7txAbZ2YP9NAxfryTwYfU14pw1Erm4X35nCrmtDYkm4b67Bq9g9dFNcCMIRVHmIcd8vkOnCPsOjTvHme1U&m=CdtiI9R0mON0i1AE-8XpA1IzBpuue2fjNUG7OFmhx-A&s=gNQKeqo4jn3JWZgEqzDqFjb4bXkq_xKLc8Lkk8aBWX0&e= >

________________________________

**Comment by hparkertt** [comment]
@jptheripper @rgwozdz @ButchGis I think this issue can be closed thanks to #162 ?
